__tests__/app.test.js

```diff
@@ -273,3 +273,73 @@ describe("POST /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("200: PATCH: responds with the updated article and the votes property should be updated by the given amount", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 20 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.votes).toEqual(120);
+        expect(body.article).toEqual(
+          expect.objectContaining({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: "2020-07-09T20:11:00.000Z",
+            votes: 120,
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          })
+        );
+      });
+  });
+  test("200: PATCH: responds with an updated article with votes incremented accordingly and ignores any other keys added into the request", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 40, likes: 300 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.votes).toEqual(140);
+      });
+  });
+  test("200: PATCH: responds with an updated article with votes number unchanged if 0 is inputted", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 0 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.votes).toEqual(100);
+      });
+  });
+  test("400: ERROR: responds with an error when article_id type is invalid", () => {
+    return request(app)
+      .patch("/api/articles/nonesense")
+      .send({ inc_votes: 40 })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("400: ERROR: responds with an error when inc_votes is blank", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({})
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("404: ERROR: responds with an error when article_id type is valid but doesn't exist", () => {
+    return request(app)
+      .patch("/api/articles/9999")
+      .send({ inc_votes: 27 })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Path not found");
+      });
+  });
```

Comment:
Really excellent testing so far Zahraa! Can I just request one more though please? Could you please add a test for when inc_votes is not a number? I.e. "inc_votes: nonsense"

Reply:
Of course, will add in now, thank you!